tests/test_phase_psf.py

```diff
@@ -399,57 +436,149 @@ def test_ne():
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.1)]
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, flux=1.1)]
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, interpolant='linear')]
     stepk = objs[0].stepK()
     maxk = objs[0].maxK()
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, _force_stepk=stepk/1.5)]
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, _force_maxk=maxk*2.0)]
     all_obj_diff(objs)
 
 
+@timer
+def test_phase_gradient_shoot():
+    # Make the atmosphere
+    seed = 12345
+    r0_500 = 0.2  # m
+    nlayers = 6
+    time_step = 0.025  # s
+    screen_size = 102.4  # m
+    screen_scale = 0.1  # m
+    max_speed = 20  # m/s
+
+    rng = galsim.BaseDeviate(seed)
+    u = galsim.UniformDeviate(rng)
+    Ellerbroek_alts = [0.0, 2.58, 5.16, 7.73, 12.89, 15.46]  # km
+    Ellerbroek_weights = [0.652, 0.172, 0.055, 0.025, 0.074, 0.022]
+    Ellerbroek_interp = galsim.LookupTable(
+            Ellerbroek_alts,
+            Ellerbroek_weights,
+            interpolant='linear')
+    alts = np.max(Ellerbroek_alts)*np.arange(nlayers)/(nlayers-1)
+    weights = Ellerbroek_interp(alts)
+    weights /= sum(weights)
+
+    spd = []  # Wind speed in m/s
+    dirn = [] # Wind direction in radians
+    r0_500s = [] # Fried parameter in m at a wavelength of 500 nm.
+    for i in range(nlayers):
+        spd.append(u()*max_speed)
+        dirn.append(u()*360*galsim.degrees)
+        r0_500s.append(r0_500*weights[i]**(-3./5))
+    atm = galsim.Atmosphere(r0_500=r0_500, speed=spd, direction=dirn, altitude=alts, rng=rng,
+                            time_step=time_step, screen_size=screen_size,
+                            screen_scale=screen_scale)
+
+    lam = 500.0
+    diam = 4.0
+    pad_factor = 1.0
+    oversampling = 1.0
+
+    aper = galsim.Aperture(diam=diam, lam=lam,
+                           screen_list=atm, pad_factor=pad_factor,
+                           oversampling=oversampling)
+
+    xs = np.empty((10,), dtype=float)
+    ys = np.empty((10,), dtype=float)
+    u.generate(xs)
+    u.generate(ys)
+    thetas = [(x*galsim.degrees, y*galsim.degrees) for x, y in zip(xs, ys)]
+
+    if __name__ == '__main__':
+        exptime = 15.0
+        centroid_tolerance = 0.05
+        second_moment_tolerance = 0.5
+    else:
+        exptime = 0.2
+        centroid_tolerance = 0.2
+        second_moment_tolerance = 1.5
+
+    psfs = [atm.makePSF(lam, diam=diam, theta=th, exptime=exptime, aper=aper) for th in thetas]
+    shoot_moments = []
+    fft_moments = []
+
+    # At the moment, Ixx and Iyy (but not Ixy) are systematically smaller in phase gradient shooting
+    # mode than in FFT mode.  For now, I'm willing to accept this, but we should revisit it once we
+    # get the "second kick" approximation implemented.
+    offset = [0.5, 0.5, 0.0]
```

Comment:
Same issue as above: the size of the image needs to be large enough to accommodate the PSF.  Then the calculated second moments make sense.